lib/puppet/network/http/api/v3.rb

```diff
@@ -14,9 +14,13 @@ class Puppet::Network::HTTP::API::V3
     Environments.new(Puppet.lookup(:environments))
   end)
 
-  def self.routes
-    Puppet::Network::HTTP::Route.path(%r{/v3}).
+  def self.master_routes
+    Puppet::Network::HTTP::Route.path(%r{v3}).
         any.
         chain(ENVIRONMENTS, INDIRECTED)
   end
+
+  def self.ca_routes
+    Puppet::Network::HTTP::Route.path(%r{v1}).any.chain(INDIRECTED)
```

Comment:
This is going to sound ridiculous but I wonder if it wouldn't make sense to go ahead and have `api/master/v3.rb` and `api/ca/v1.rb` now.  It's ridiculous because those namespaces would only have like 3 lines of code in them, but I kind of think it might still be worth it because otherwise the `v3.rb` file doesn't seem like an intuitive place to look for the v1 CA routes... :(  I could survive with it like this but I think I might have a slight preference for going ahead and breaking it out now.